src/plugins/intel_cpu/src/utils/jit_kernel.hpp

```diff
@@ -0,0 +1,697 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include <cpu/x64/jit_generator.hpp>
+#include <emitters/jit_load_store_emitters.hpp>
+#include <ie/ie_precision.hpp>
+#include <common/nstl.hpp>
+#include <type_traits>
+#include <functional>
+#include <vector>
+#include <array>
+#include <tuple>
+
+namespace MKLDNNPlugin {
+
+struct jit_kernel;
+
+namespace internal {
+
+template<size_t S>
+struct reg_traits_by_size;
+template<typename T>
+struct reg_traits;
+template<typename T, size_t N>
+struct reg_traits<T[N]>;
+template<dnnl::impl::cpu::x64::cpu_isa_t isa>
+struct isa_traits;
+
+template<>
+struct reg_traits_by_size<1> {
+    using type = Xbyak::Reg8;
+    constexpr static size_t size = 1;           // in bytes
+    constexpr static dnnl::impl::cpu::x64::cpu_isa_t isa
+                        = dnnl::impl::cpu::x64::cpu_isa_t::isa_any;
+};
+
+template<>
+struct reg_traits_by_size<2> {
+    using type = Xbyak::Reg16;
+    constexpr static size_t size = 2;           // in bytes
+    constexpr static dnnl::impl::cpu::x64::cpu_isa_t isa
+                        = dnnl::impl::cpu::x64::cpu_isa_t::isa_any;
+};
+
+template<>
+struct reg_traits_by_size<4> {
+    using type = Xbyak::Reg32;
+    constexpr static size_t size = 4;           // in bytes
+    constexpr static dnnl::impl::cpu::x64::cpu_isa_t isa
+                        = dnnl::impl::cpu::x64::cpu_isa_t::isa_any;
+};
+
+template<>
+struct reg_traits_by_size<8> {
+    using type = Xbyak::Reg64;
+    constexpr static size_t size = 8;           // in bytes
+    constexpr static dnnl::impl::cpu::x64::cpu_isa_t isa
+                        = dnnl::impl::cpu::x64::cpu_isa_t::isa_any;
+};
+
+template<typename T>
+struct reg_traits : public reg_traits_by_size<sizeof(T)> {};
+
+template<>
+struct reg_traits<float> {
+    using type = Xbyak::Fpu;
+    constexpr static size_t size = 10;          // in bytes
+    constexpr static dnnl::impl::cpu::x64::cpu_isa_t isa
+                        = dnnl::impl::cpu::x64::cpu_isa_t::isa_any;
+};
+template<>
+struct reg_traits<double> : public reg_traits<float> {};
+
+template<typename T>
+struct reg_traits<T[1]> : public reg_traits<T> {};
+
+template<typename T>
+struct reg_traits<T[4]> {
+    using type = Xbyak::Xmm;
+    constexpr static size_t size = 4 * 4;       // in bytes
+    constexpr static dnnl::impl::cpu::x64::cpu_isa_t isa
+                        = dnnl::impl::cpu::x64::cpu_isa_t::sse41;
+};
+
+template<typename T>
+struct reg_traits<T[8]> {
+    using type = Xbyak::Ymm;
+    constexpr static size_t size = 8 * 4;       // in bytes
+    constexpr static dnnl::impl::cpu::x64::cpu_isa_t isa
```

Comment:
not sure what this isa actually means and how we expect it to be used. We can have Ymm (as half of Zmm) and avx512_isa combination.

Reply:
This bug is fixed by PR 9118. Here isa is required to determine the minimum ISA where the appropriate register is defined.